geth/jail/execution_policy.go

```diff
@@ -3,148 +3,83 @@ package jail
 import (
 	"context"
 	"encoding/json"
-	"math/big"
-	"time"
 
 	gethcommon "github.com/ethereum/go-ethereum/common"
-	"github.com/ethereum/go-ethereum/common/hexutil"
-	"github.com/ethereum/go-ethereum/core/types"
-	"github.com/ethereum/go-ethereum/rlp"
 	"github.com/ethereum/go-ethereum/rpc"
 	"github.com/robertkrimen/otto"
 	"github.com/status-im/status-go/geth/common"
+	"github.com/status-im/status-go/geth/params"
 )
 
 // ExecutionPolicy provides a central container for the executions of RPCCall requests for both
 // remote/upstream processing and internal node processing.
 type ExecutionPolicy struct {
 	nodeManager    common.NodeManager
 	accountManager common.AccountManager
+	txQueueManager common.TxQueueManager
 }
 
 // NewExecutionPolicy returns a new instance of ExecutionPolicy.
-func NewExecutionPolicy(nodeManager common.NodeManager, accountManager common.AccountManager) *ExecutionPolicy {
+func NewExecutionPolicy(
+	nodeManager common.NodeManager, accountManager common.AccountManager, txQueueManager common.TxQueueManager,
+) *ExecutionPolicy {
 	return &ExecutionPolicy{
 		nodeManager:    nodeManager,
 		accountManager: accountManager,
+		txQueueManager: txQueueManager,
 	}
 }
 
-// ExecuteSendTransaction defines a function to execute RPC requests for eth_sendTransaction method only.
-func (ep *ExecutionPolicy) ExecuteSendTransaction(req common.RPCCall, call otto.FunctionCall) (*otto.Object, error) {
-	config, err := ep.nodeManager.NodeConfig()
-	if err != nil {
-		return nil, err
-	}
-
-	if config.UpstreamConfig.Enabled {
-		return ep.executeRemoteSendTransaction(req, call)
+// Execute handles a received RPC call.
+func (ep *ExecutionPolicy) Execute(req common.RPCCall, call otto.FunctionCall) (*otto.Object, error) {
+	switch req.Method {
+	case params.SendTransactionMethodName:
+		return ep.executeSendTransaction(req, call)
+	default:
+		return ep.executeOtherTransaction(req, call)
```

Comment:
You might need to fix conflict merges if #276  gets merged before this PR.

Reply:
Yup, I am prepared for that :)